Homework-06-documentation-tests.ipynb

```diff
@@ -54,7 +90,20 @@
     "text = \"Betty Botter bought some butter. But she said the butter’s bitter. If I put it in my batter, it will make my batter bitter. But a bit of better butter will make my batter better. So it was better Betty Botter bought a bit of better butter.\"\n",
     "\n",
     "stat = TextStats(text)\n",
-    "print(stat.count_words())\n"
+    "stat_fin = (stat.count_words())\n",
+    "print(stat_fin)\n",
+    "\n",
+    "\n",
+    "import unittest\n",
+    "\n",
+    "#test case: this test cases makes sure the counter works by comparing the result to the correct count.\n",
+    "class TestCountWords(unittest.TestCase):\n",
+    "    \n",
+    "    def test_count_words(self):\n",
+    "        expected_result = {'betty': 2, 'botter': 2, 'bought': 2, 'some': 1, 'butter': 3, 'but': 2, 'she': 1, 'said': 1, 'the': 1, 'butter’s': 1, 'bitter': 2, 'if': 1, 'i': 1, 'put': 1, 'it': 3, 'in': 1, 'my': 3, 'batter': 3, 'will': 2, 'make': 2, 'a': 2, 'bit': 2, 'of': 2, 'better': 4, 'so': 1, 'was': 1}\n",
+    "        self.assertEqual(stat_fin, expected_result)\n",
+    "\n",
+    "unittest.main(argv=[''], exit=False)"
```

Comment:
your test method needs to call the method it is testing. So in your test method you would create a string, call the method and then compare that output with what it should be. Test cases should be independent of the rest of your code.

Reply:
also, you would test all possible scenarios, e.g. the string has no words, just one word, multiple words.